File: frontend/web/src/components/conversation/persona.ts
import { AGENT_KEYS, type AgentKey } from '@/types/conversation'

/** Display metadata for an Aveline agent persona. */
export interface Persona {
  key: string
  name: string
  /** Tailwind text color class for the persona accent. */
  text: string
  /** Tailwind background color class for the persona accent. */
  bg: string
  /** Tailwind soft (tinted) background color class for avatar chips. */
  bgSoft: string
  /** Tailwind ring/border color class. */
  ring: string
  /** Short descriptor shown under the name. */
  role: string
}

const AGENT_PERSONAS: Record<AgentKey, Persona> = {
  [AGENT_KEYS.aveline]: {
    key: AGENT_KEYS.aveline,
    name: 'Aveline',
    text: 'text-primary',
    bg: 'bg-primary',
    bgSoft: 'bg-primary/10',
    ring: 'ring-primary/20',
    role: 'Your concierge',
  },
  [AGENT_KEYS.ava]: {
    key: AGENT_KEYS.ava,
    name: 'Ava',
    text: 'text-memory',
    bg: 'bg-memory',
    bgSoft: 'bg-memory/10',
    ring: 'ring-memory/20',
    role: 'Memory',
  },
  [AGENT_KEYS.elle]: {
    key: AGENT_KEYS.elle,
    name: 'Elle',
    text: 'text-visual',
    bg: 'bg-visual',
    bgSoft: 'bg-visual/10',
    ring: 'ring-visual/20',
    role: 'Visual sourcing',
  },
  [AGENT_KEYS.lina]: {
    key: AGENT_KEYS.lina,
    name: 'Lina',
    text: 'text-commerce',
    bg: 'bg-commerce',
    bgSoft: 'bg-commerce/10',
    ring: 'ring-commerce/20',
    role: 'Commerce',
  },
}

/** Resolves a persona for an agent key, defaulting to Aveline for unknown keys. */
export function personaForAgent(agentKey: string | null | undefined): Persona {
  if (agentKey && agentKey in AGENT_PERSONAS) {
    return AGENT_PERSONAS[agentKey as AgentKey]
  }
  return AGENT_PERSONAS[AGENT_KEYS.aveline]
}

/** Resolves a persona for a message author kind + agent key. */
export function personaForAuthor(
  authorKind: string,
  agentKey: string | null | undefined,
): Persona | null {
  if (authorKind === 'Agent') {
    return personaForAgent(agentKey)
  }
  return null
}

File: frontend/aveline_mobile/lib/shared/persona.dart
import 'package:flutter/material.dart';

/// Display metadata for an Aveline agent persona, mirroring the web
/// `persona.ts`. Accent colors follow `.agents/brain/DESIGN.md`.
class Persona {
  const Persona({
    required this.name,
    required this.accent,
    required this.role,
  });

  final String name;
  final Color accent;
  final String role;
}

/// Aveline's primary wine-rose.
const Color _avelineAccent = Color(0xFF8B2E42);

/// Memory's muted magenta-rose.
const Color _avaAccent = Color(0xFFB0566B);

/// Visual's soft gold/brass.
const Color _elleAccent = Color(0xFFB08D57);

/// Commerce's lilac.
///
/// Lina used to share Aveline's wine-rose (`0xFF8B2E42`), which is also the theme seed and the
/// colour the docs chrome and the pricing CTA are built on — so her avatar was indistinguishable
/// from the app's own accent. She now has her own lilac, matching `--aveline-lilac` in the web
/// palette. It is deliberately deeper than the decorative lavender (`0xFF8E7CC3`) because her
/// name label and avatar glyphs sit on this colour at 11px, where lavender is only ~3.6:1 against
/// white; this clears WCAG AA at ~5.9:1.
const Color _linaAccent = Color(0xFF6B5A9E);

const Map<String, Persona> _personas = {
  'aveline': Persona(name: 'Aveline', accent: _avelineAccent, role: 'Your concierge'),
  'ava': Persona(name: 'Ava', accent: _avaAccent, role: 'Memory'),
  'elle': Persona(name: 'Elle', accent: _elleAccent, role: 'Visual sourcing'),
  'lina': Persona(name: 'Lina', accent: _linaAccent, role: 'Commerce'),
};

/// Resolves a persona for an agent key, defaulting to Aveline for unknown keys.
Persona personaForAgent(String? agentKey) {
  if (agentKey != null && _personas.containsKey(agentKey)) {
    return _personas[agentKey]!;
  }
  return _personas['aveline']!;
}

/// Resolves a persona for a message author kind + agent key, or `null` when the
/// author is not an agent.
Persona? personaForAuthor(String authorKind, String? agentKey) {
  if (authorKind == 'Agent') {
    return personaForAgent(agentKey);
  }
  return null;
}
